idega/fake-backend/infra/WineRepository.ts

```diff
@@ -0,0 +1,42 @@
+import fs from 'fs';
+import { join, dirname } from 'node:path';
+import { fileURLToPath } from 'node:url';
+
+import { Wine } from '../domain/Wines';
+import { WineRepository } from '../domain/wineRepository';
+
+import inMemoryDb from '../data/inMemoryDb';
+
+const _dirname = dirname(fileURLToPath(import.meta.url));
+const file = join(_dirname, '..', 'data', 'db.json');
+
+let data = process.env.NODE_ENV !== 'test' ? JSON.parse(fs.readFileSync(file, { encoding: 'utf8' })) : inMemoryDb();
+
+export const wineRepository: WineRepository = {
+  getNextId: () => (data.wines.length ? Math.max(...data.wines.map((x: Wine.Type) => x.id)) + 1 : 1),
```

Comment:
```suggestion
  getNextId: () => (data.wines.length ? Math.max(...data.wines.map((wine: Wine.Type) => wine.id)) + 1 : 1),
```
Just for better understanding of the code